cni/network/network.go

```diff
@@ -169,13 +165,17 @@ func (plugin *netPlugin) Add(args *cniSkel.CmdArgs) error {
 		addSnatInterface(nwCfg, result)
 
 		// Convert result to the requested CNI version.
-		res, err := result.GetAsVersion(nwCfg.CNIVersion)
-		if err != nil {
-			err = plugin.Error(err)
+		res, vererr := result.GetAsVersion(nwCfg.CNIVersion)
+		if vererr != nil {
+			log.Printf("GetAsVersion failed with error %v", vererr)
+			plugin.Error(vererr)
+		}
+
+		if err == nil {
```

Comment:
if err==nil && res!=nil

asking because if GetAsVersion fails, do we still want to print to stdout?